### PR TITLE
CORE-5001

### PR DESCRIPTION
**Pull request by JReissmueller**
Implemented getDomainNameServers and setDomainNameServers

### CODE REVIEW

File: openprovider.php [hunk 1]
```diff
@@ -2274,6 +2274,96 @@ public function getServiceDomain($service): string
...
+     * Gets a list of name server data associated with a domain
+     *
+     * @param string $domain The domain to lookup
+     * @param int $module_row_id The ID of the module row to fetch for the current module
+     * @return array A list of name servers, each with the following fields:
+     *
+     *  - url The URL of the name server
+     *  - ips A list of IPs for the name server
+     */
+    public function getDomainNameServers($domain, $module_row_id = null)
```
Comment (abdyfranco) [inline openprovider.php:2286]: It seems this method is not fully working, The most recent update of the Domains plugin calls this method to fill the name servers fields in the Manage view, but it does not appear to work:
Screenshot 2023-11-20 at 3 12 56 PM

File: openprovider.php [hunk 2]
```diff
@@ -2274,6 +2274,96 @@ public function getServiceDomain($service): string
...
+
+    /**
+     * Assign new name servers to a domain
+     *
+     * @param string $domain The domain for which to assign new name servers
+     * @param int|null $module_row_id The ID of the module row to fetch for the current module
+     * @param array $vars A list of name servers to assign (e.g. [ns1, ns2])
+     * @return bool True if the name servers were successfully updated, false otherwise
+     */
+    public function setDomainNameservers($domain, $module_row_id = null, array $vars = [])
```
Comment (abdyfranco) [inline openprovider.php:2329]: I think the "Nameservers" tab (method manageNameservers) should use these new methods, as the current one appears to be broken. I get the following error trying to update the nameservers:

`Uncaught TypeError, code 0 in /Volumes/Archive/Documents/Projects/GitHub/module-openprovider/apis/openprovider_api.php on line 184 Message: Return value of OpenProviderApi::getLastRequest() must be an instance of LastRequest, null returned`

**Comment by abdyfranco** [comment]
These are some comments not related to this PR, but related to the changes made in the initial fork of the module:
 - The title on the README.md file has "Internet.bs" instead of "OpenProvider"
 - Views still make use of the $this->Html->ifSet() method, method that has been deprecated in favor of better compatibility with PHP 8.0
 - On file openprovider.php at line 68, The module tries to set the module for the first occurrence from ModuleManager::getInstalled(), without passing a company_id filter. This may broke functionality on a mutli-company setup as it may set a module from another company as the current module. I believe this piece of code is not necessary as the ModuleManager already do this in a proper way.
 - The PHPDoc comment on the getPrices() method, mentions Namesilo instead of OpenProvider
 - There is a debug() method that doesn't seem to be in use, I think it should be removed like we did with Namesilo in the past
 - The fields on the admin "Contacts" tab, have the "input-block-level w-100" classes making the UI inconsistent with the rest of the system.

**Review by abdyfranco — CHANGES_REQUESTED** [review]

File: openprovider.php (hunk 1 above)
Comment (JReissmueller) [reply openprovider.php:2286]: Did you set nameservers?  It works for me
openprovider-nameservers

**Review by abdyfranco — APPROVED** [review]

**Pull request merged**